website/pages/docs/developers/creating-new-plugin.md

```diff
@@ -1,18 +1,388 @@
+
+With the columns defined, we can now write the resolver function. The resolver function is responsible for fetching the data from the API and returning it to CloudQuery. The resolver function takes a `context.Context` object, a `schema.ClientMeta` object, a `*schema.Resource` object, and a `chan<- interface{}` object. The `context.Context` object is used to cancel the resolver function if the user cancels the sync. The `schema.ClientMeta` object is a generic object that can be used to store any data that needs to be shared between resolvers. The `*schema.Resource` object is the parent resource of the table, if any, and is used to implement parent-child relationships. In our case this will be `nil`. The `chan<- interface{}` object is used to send the data back to CloudQuery.
+
+We won't go into all the details of making API calls here, but let's look at what a resolver function for our XKCD plugin might look like:
+
+```go
+
+func fetchComics(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	client := meta.(*Client)
+	latest, err := client.XKCD.GetLatestComic(ctx)
+	if err != nil {
+		return err
+	}
+	res <- latest
+
+	for i := 1; i < latest.Num; i++ {
+		comic, err := client.XKCD.GetComic(ctx, i)
+		if err != nil {
+			return err
+		}
+		res <- comic
+	}
+	return nil
+}
+```
+
+In the above code, we are getting a list of Comics from the XKCD API and sending them to the CloudQuery over the `res` channel. We first need to get the latest comic, then we can iterate through all the IDs from 1 to that number. You can send items to the channel one at a time, or as a slice of items. The sooner an is dispatched over the channel, the sooner it will be written to the destination(s), so we prefer to write them as soon as they are available. And as long as the struct sent matches the one used for the table, the CloudQuery SDK will handle the rest.
+
+In the above example, we used a `Client` struct that we haven't talked about yet. The `Client` struct is used to store any data that needs to be shared between resolvers. For example, it may store the API key that we'll need to make API calls, or an SDK client that we'll use to make API calls. The `Client` struct is defined in the `client` directory, and is instantiated with a call to `client.New` in the `plugin` directory. In this case, we were using it to store an instance of the XKCD client. (We won't show the full XKCD client implementation here.)
+ 
+### Testing the Plugin
+
+There are two options for running a plugin before as a developer before it is released: as a gRPC server, or as a standalone binary.
+
+#### Run the Plugin as a gRPC Server
+
+This mode is especially useful for setting breakpoints your code for debugging, as you can run it in server mode from your IDE and attach a debugger to it. To run the plugin as a gRPC server, you can run the following command in the root of the plugin directory:
+
+```shell
+go run main.go serve
```

Comment:
I always do `go run . serve` which I find easier to write (or search in my bash history)